snake.asm

```diff
@@ -1,42 +1,42 @@
-push 0xB800
-pop ds
-mov cx, 0xFA0
-std
-start:
-	mov ax, 0x3
-	int 0x10
-	mov di, 0x7D0
-	mov si, sp
-.food:
-	imul bx, sp
-	and bx, cx
-	cmp [bx], ch
-	je .food
-	mov [bx], cl
-.input:
-	in al, 0x60
-	mov bx, 0x4
-	and al, 0x1E
-	jp $+0x4
-	mov bl, cl
-	and al, 0x14
-	jz $+0x4
-	neg bx
-	sub di, bx
-	cmp di, cx
-	ja start
-	sar bx, 0x1
-	lea ax, [di+bx+0x2]
-	div cl
-	and ah, ah
-	jz start
-	cmp [di], ch
-	je start
-	push di
-	cmp [di], cl
-	mov [di], ch
-	je .food
-	es lodsw
-	xchg ax, bx
-	mov [bx], ah
-	jmp SHORT .input
+push 0xB800                 ; push the offset to video memory for DMA
+pop ds                      ; pop that value into the data segment register so that every memory access will edit the video memory
+mov cx, 0xFA0               ; move the size of the screen in text mode 3 into cx, also the cp437 values of the food and snake body into ch, cl, and the width of the screen to cl

```

Comment:
Might be worth putting a multiline comment on top of this line since there's a lot going on here. Maybe explain how 0xfa0 is broken down in multiple pieces and going into ch and cl.